internal/serverutil/serverutil.go

```diff
@@ -0,0 +1,69 @@
+package serverutil
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/certwatcher"
+
+	"github.com/operator-framework/catalogd/internal/third_party/server"
+	catalogdmetrics "github.com/operator-framework/catalogd/pkg/metrics"
+	"github.com/operator-framework/catalogd/pkg/storage"
+)
+
+type CatalogServerConfig struct {
+	ExternalAddr string
+	CatalogAddr  string
+	CertFile     string
+	KeyFile      string
+	LocalStorage storage.Instance
+}
+
+func AddCatalogServerToManager(mgr ctrl.Manager, cfg CatalogServerConfig) error {
+	listener, err := net.Listen("tcp", cfg.CatalogAddr)
+	if err != nil {
+		return err
```

Comment:
Could we wrap all the errors returned by this function with some more context? For example, for this error I'd like to see something along the lines of:
```suggestion
		return fmt.Errorf("creating listener for %q: %w", cfg.CatalogAddr, err)
```

This will help with troubleshooting any problems since we would be able to more easily tell from the logs where exactly in this function the error was returned

Reply:
done, adjusted the top-level error in main.go as well